src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioManagedModuleUpdateProvider.cs

```diff
@@ -35,7 +39,7 @@ public async Task<bool> HasChangesAsync(string sourceFilePath, CancellationToken
         {
             try
             {
-                return await _encService.HasChangesAsync(sourceFilePath, cancellationToken).ConfigureAwait(false);
+                return await _encService.HasChangesAsync(_workspace.CurrentSolution, sourceFilePath, cancellationToken).ConfigureAwait(false);
```

Comment:
seems slightly weird to just grab whatever is 'CurrentSOlution' in an async method.  Is htat desirable here?  Would t be cleaner/clearer to have the topmost caller grab this snapshot and pass it along so that every called method has a consistent view of what tehy're answering questions about?

Reply:
This is the top-most caller. The debugger does not have any knowledge about solutions it is querying about the current state.